### PR TITLE
VSCode server path use compiler path instead of bin

### LINKED ISSUE #40: Simplify specifying cadl-server location in VS Code settings
Today you have to specify a path to a cadl-server(.cmd) executable. This is hard for two reasons: 

1. Even if you have that, it's deep and hard to find
2. If you want to run against a local build of the package, as we do, you don't even have that. 

Our workaround for (2) to use the .bin/cadl-server in packages/samples that consumes packages/compiler.

It would be better if you could just specify the package path and it worked.

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #40

Simplify redefining where to load `cadl-server` from. This should just be the path to the `@cadl-lang/compiler` package.
This is more similar to how typescript customize its path

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,16 +12,16 @@ using the preview.
```
Comment (timotheeguerin) [inline README.md:12]: Guess prettier just formatted  in vscode, let me know if we don't want to do that.

**Comment by bterlson** [comment]
What does this change in terms of end experience? (Very clueless when it comes to VSC extensions).

**Comment by timotheeguerin** [comment]
> What does this change in terms of end experience? (Very clueless when it comes to VSC extensions).

It is just a little more obvious to the user what the path correspond to now and for use case like our own(Not sure anybody else will have it ever) that path can be much more meaningful (Path to the compiler code vs the workaround we had of the binary installed in the samples package)

File: packages/cadl-vscode/src/extension.ts [hunk 2]
```diff
@@ -71,7 +61,18 @@ function resolveCadlServer(context: ExtensionContext): Executable {
...
-  return { command, args, options };
+  // Default to cadl-server on PATH, which would come from `npm install -g
+  // @cadl-lang/compiler` in a vanilla setup.
+  if (!serverPath) {
+    const executable = process.platform === "win32" ? "cadl-server.cmd" : "cadl-server";
+    return { command: executable, args, options };
+  }
+
+  if (!serverPath.endsWith(".js")) {
+    serverPath = join(serverPath, "cmd/cadl-server.js");
```
Comment (nguerrera) [inline packages/cadl-vscode/src/extension.ts:72]: I wonder if we can make this support all cases and stay compatible. 

If ends with .js, run with node
If not ends with .js and (path)/cmd/cadl-server.js exists, use that.
Else run as executable

In addition to being non-breaking, it gives you an escape hatch if you need to run with a node not on path or whatever, you could always craft your own executable.

Not really sure this is necessary, but thought I'd throw it out there.

**Review by nguerrera — APPROVED** [review]
Thanks!

**Pull request merged**